template.yaml

```diff
@@ -112,13 +126,29 @@ Resources:
                   - logs:PutLogEvents
                 Resource: '*'
 
+  DatabaseAccessLambdaPermissions:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: !Sub DatabaseAccessLambdaPermissions-${AWS::StackName}
+      Roles:
+        - !Ref PreferencesLambdaRole
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Action:
+              - dynamodb:*
+            Resource:
+              - !Sub arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/nva-person-preferences-${AWS::StackName}
+              - !Sub arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/nva-person-preferences-${AWS::StackName}/index/*
+
   #============================= Handlers ==========================================================
 
-  NvaCreateProfileHandler:
+  NvaCreatePreferencesHandler:
     Type: AWS::Serverless::Function
     Properties:
```

Comment:
Try to always add a short but descriptive `Description` under `Properties` in template.